pkg/controllers/controlplanemachinesetgenerator/controller_test.go

```diff
@@ -1970,6 +1934,28 @@ var _ = Describe("controlplanemachinesetgenerator controller on Nutanix", func()
 	var cpms *machinev1.ControlPlaneMachineSet
 	var machine0, machine1, machine2 *machinev1beta1.Machine
 	var machineSet0, machineSet1, machineSet2, machineSet3, machineSet4 *machinev1beta1.MachineSet
+	var withNutanixFailureDomains bool
+
+	createInfrastructure := func() *configv1.Infrastructure {
+		By("Setting up a new infrastructure for the test")
+
+		infra := util.CreateNewInfrastructureObject4NutanixTest(withNutanixFailureDomains)
+		Expect(infra != nil && infra.Spec.PlatformSpec.Nutanix != nil).To(BeTrue())
+		infraStatus := infra.Status.DeepCopy()
+		Expect(k8sClient.Create(ctx, infra)).To(Succeed())
+		// Update Infrastructure Status.
+		Eventually(komega.UpdateStatus(infra, func() {
+			infra.Status = *infraStatus
+		})).Should(Succeed())
+
+		if withNutanixFailureDomains {
+			Expect(len(infra.Spec.PlatformSpec.Nutanix.FailureDomains) == 3).To(BeTrue())
```

Comment:
Using the actual assertions rather than `BeTrue()` will be easier to debug
```suggestion
			Expect(infra.Spec.PlatformSpec.Nutanix.FailureDomains).To(HaveLen(3), "Expected 3 failure domains for the nutanix platform spec")
```

Reply:
Changed